docker/bitcoind/Dockerfile

```diff
@@ -3,17 +3,25 @@ FROM ubuntu:24.04
 
 WORKDIR /app
 
+# Set bitcoind version and architecture as build arguments
+ARG BITCOIND_VERSION="28.0"
+ARG BITCOIND_ARCH="x86_64-linux-gnu"
+ARG SHASUM="7fe294b02b25b51acb8e8e0a0eb5af6bbafa7cd0c5b0e5fcbb61263104a82fbc"
+
 # Install necessary dependencies and add PPA
 RUN apt-get update && \
     apt-get install -y software-properties-common && \
-    add-apt-repository ppa:luke-jr/bitcoincore && \
-    apt-get update && \
-    apt-get install -y \
-    bitcoind \
     curl && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
+# Download, verify, and install bitcoind
+RUN curl -fsSLO --proto '=https' --tlsv1.2 \
+    "https://bitcoincore.org/bin/bitcoin-core-$BITCOIND_VERSION/bitcoin-$BITCOIND_VERSION-$BITCOIND_ARCH.tar.gz" && \
+    echo "$SHASUM bitcoin-$BITCOIND_VERSION-$BITCOIND_ARCH.tar.gz" | sha256sum -c - && \
+    tar xzf "bitcoin-$BITCOIND_VERSION-$BITCOIND_ARCH.tar.gz" && \
+    install -m 0755 -t /usr/local/bin bitcoin-$BITCOIND_VERSION/bin/* && \
+    rm -rf "bitcoin-$BITCOIND_VERSION-$BITCOIND_ARCH.tar.gz" "bitcoin-$BITCOIND_VERSION"
```

Comment:
I need people who made the docker changes in the past to vouch for these changes.
Cc @voidash, @Rajil1213 or @sapinb

Reply:
If building this works, then it's fine I think. I can't see anything wrong with the changes themselves.

Although if you _are_ specifying the architecture, docker-compose file might need to change as well because not all of us are working on `x86-64` archs.

Reply:
Can you check if a455dc6992052851f8244d97dc8f2923c194611a works?
I don't know how to this elegantly.

Reply:
Because `apt` is not an official repo. The only official distribution channel is downloading the binaries and verifying PGP sigs and SHA hashes.

Reply:
https://bitcoincore.org/en/download/

Reply:
That might not matter much because the one in devnet is using a different version (I think [this one](https://github.com/MutinyWallet/mutiny-net)). This Dockerfile is mostly for running local, manual tests. So, I don't know if the amount of work required to make this cross platform is worth the effort.

Reply:
Cc @purusang